### PR TITLE
More multiple region support.

### PR DESCRIPTION
**Pull request by leftwo**
Even more updates to support Volume layer activities.

Added a new test to CI that will run "test_up.sh encrypted" with two region sets (6 downstairs).

Crutest changes:
Added some comments to crutest tests that were missing them.
Crutest will now determine the total number of downstairs it expects to be using and compare that value when looking for an expected number of downstairs in an active state.
Updated replay, replace, replace-before-active, and replace-while-reconcile tests to support having more than one region set.
Fixed a bug in the replace-before-active test where previous tests could leave things broken in a way the test could not repair.

Removed the no longer valid NEW columns from the DTrace `single_up_info.d` script.

test_up.sh changes:
Added support for running all the tests with a user supplied number of region sets instead of just defaulting to one (three downstairs).
Some other misc cleanup of test_up.sh.

### CODE REVIEW

**Comment by leftwo** [comment]
I believe I've found all the problems CI was tripping over.. finally!

**Comment by leftwo** [comment]
Fixing https://github.com/oxidecomputer/crucible/issues/1491 was part of the changes done here as well.

File: crutest/src/main.rs [hunk 1]
```diff
@@ -2431,15 +2471,15 @@ async fn replace_while_reconcile(
...
+            if wc.up_count + wc.ds_count == 0 && wc.active_count == ds_total {
+                info!(log, "[{c}] All jobs finished, all DS active.");
                 break;
             }
             tokio::time::sleep(tokio::time::Duration::from_secs(4)).await;
         }
 
-        old_ds = (old_ds + 1) % 4;
-        new_ds = (new_ds + 1) % 4;
+        old_ds = (old_ds + 1) % (ds_total as u32 + 1);
```
Comment (mkeeter) [inline crutest/src/main.rs:2481]: Nit: what's with the `as u32` cast here (and below)?

File: tools/test_up.sh [hunk 2]
```diff
@@ -212,8 +237,8 @@ echo cp -r "${testdir}/${port}" "${testdir}/previous"
 cp -r "${testdir}/${port}" "${testdir}/previous"
 
 new_args+=( --verify-in "${testdir}/verify_file" )
-echo "$ct" repair -g "$gen" -q "${new_args[@]}"
-if ! "$ct" repair -g "$gen" -q "${new_args[@]}"; then
+echo "$ct" fill -g "$gen" -q "${new_args[@]}"
+if ! "$ct" fill -g "$gen" -q "${new_args[@]}"; then
```
Comment (mkeeter) [inline tools/test_up.sh:241]: Is this change intentional?

Comment (leftwo) [reply tools/test_up.sh:241]: Yes, this was one of the test bugs.
Sometimes the repair test would not actually change the downstairs that we dump below.
The `fill` touches every extent, so we know there should be a difference when we dump.

File: crutest/src/main.rs (hunk 1 above)
Comment (leftwo) [reply crutest/src/main.rs:2481]: I'm just doing what the compiler tells me, man!

`old_ds` which is sent to dsc_client functions wants it to be u32.
However, when we use it as an index for `targets[`,  both `old_ds` and `new_ds` need to be usize.

If you think it's better, I could make it a `usize`, and cast to` u32` when I send it to dsc?

**Review by mkeeter — APPROVED** [review]
